components/Starknet/modules/architecture_and_concepts/pages/Accounts/deploying_new_accounts.adoc

```diff
@@ -1,61 +1,74 @@
 
-== Validate deploy
+* Restricted access to `+sequencer_address+` in the `+get_execution_info+` syscall. The syscall returns zero values for `+sequencer_address+`
+* Restricted access to the following syscalls:
+** `+get_block_hash+` for Cairo contracts
+** `+get_sequencer_address+` for Cairo 0 contracts
 
-Two issues might arise from sending a `deploy_account` transaction without any extra validation:
+[#validate_deploy]
+== Validating a `DEPLOY_ACCOUNT` transaction
 
-* Sequencers having the ability to charge arbitrarily high fees, thus potentially draining user funds from a pre-funded account
-* A bad actor having the ability to carry out a sequencer DOS attack by sending multiple, invalid `deploy_account` transactions. This would result in the sequencer not being compensated for work completed.
+Two potential issues with sending a `DEPLOY_ACCOUNT` transaction without any validation are:
 
+* Sequencers can charge arbitrarily high fees, potentially draining user funds from a pre-funded account.
+* A bad actor can carry out a Denial of Service (DoS) attack on the sequencer by sending multiple, invalid `DEPLOY_ACCOUNT` transactions, which would result in the sequencer not being compensated for work completed.
 
-To prevent the scenario described in the first point, a new optional validation entrypoint is provided: `+__validate_deploy__+`.
+You can prevent both issues by using a validation entrypoint called `+__validate_deploy__+`.
 
-To prevent the potential DOS attack from the second point, we introduce some limitations on the constructor and `+__validate_deploy__+`
-_ executions, namely:
+`+__validate_deploy__+` prevents sequencers from charging arbitrarily high fees, and the following limitations on the constructor and `+__validate_deploy__+` executions prevent a potential DoS attack:
 
-* Limited # of Cairo steps
-* Limited # of builtin applications
-* No external contract calls (library calls and self-calls are allowed)
+* Limited number of Cairo steps.
+* Limited number of times a builtin can be applied.
+* External contract calls are not allowed. Library calls and self-calls are allowed.
 
-This entrypoint should be included in any accounts or contracts that wish to allow this new deployment flow.
+Include the `+__validate_deploy__+` entrypoint in any accounts or contracts that can enable deploying a new account.
 
-=== Using validate deploy
+== `+__validate_deploy__+`
 
-`+__validate_deploy__+` expects the following arguments:
+Validates a `DEPLOY_ACCOUNT` transaction.
 
-* class hash
-* contract address salt
-* constructor arguments - the arguments expected by the contract’s constructor (this will be enforced in the compiler).
+[discrete]
+=== Parameters
+
+[horizontal,labelwidth="35",role="stripes-odd"]
+`class_hash: _felt_`:: The class hash.
+`contract_address_salt: _felt_`:: The contract address salt.
+`<__constructor_arguments__>`:: The arguments expected by the contract’s constructor. The compiler enforces this constructor.
```

Comment:
The compiler enforces that the inputs to the constructor are identical to the rest of the inputs for validate_deploy.